test/extensions/transport_sockets/internal_upstream/internal_upstream_integration_test.cc

```diff
@@ -182,5 +191,399 @@ TEST_F(InternalUpstreamIntegrationTest, BasicFlowWithoutTransportSocket) {
+        return new Buffer::WatermarkBuffer(below_low, above_high, above_overflow);
+      }));
+  config_helper_.renameListener("tcp_proxy");
+
+  // Setup factories and contexts for upstream clear-text raw buffer transport socket.
+  auto config = std::make_unique<envoy::extensions::transport_sockets::raw_buffer::v3::RawBuffer>();
+
+  auto factory =
+      std::make_unique<Extensions::TransportSockets::RawBuffer::UpstreamRawBufferSocketFactory>();
+  cleartext_context_ = Network::UpstreamTransportSocketFactoryPtr{
+      factory->createTransportSocketFactory(*config, factory_context_)};
+
+  // Setup factories and contexts for tls transport socket.
+  tls_context_manager_ =
+      std::make_unique<Extensions::TransportSockets::Tls::ContextManagerImpl>(timeSystem());
+  tls_context_ = Ssl::createClientSslTransportSocketFactory({}, *tls_context_manager_, *api_);
+  payload_reader_ = std::make_shared<WaitForPayloadReader>(*dispatcher_);
+
+  BaseIntegrationTest::initialize();
+
+  Network::Address::InstanceConstSharedPtr address =
+      Ssl::getSslAddress(version_, lookupPort("tcp_proxy"));
+  conn_ = std::make_unique<ClientTestConnection>(
+      *dispatcher_, address, Network::Address::InstanceConstSharedPtr(),
+      cleartext_context_->createTransportSocket(
+          std::make_shared<Network::TransportSocketOptionsImpl>(
+              absl::string_view(""), std::vector<std::string>(), std::vector<std::string>()),
+          nullptr),
+      nullptr);
+
+  conn_->enableHalfClose(true);
+  conn_->addConnectionCallbacks(connect_callbacks_);
+  conn_->addReadFilter(payload_reader_);
+}
+
+//*************************************************************************
+//  Test utility functions used by internal listener integration test
+//*************************************************************************
+
+// Setup DNS cache config.
+void HttpsInspectionIntegrationTest::setupDnsCacheConfig(
+    envoy::extensions::common::dynamic_forward_proxy::v3::DnsCacheConfig&
+    dns_cache_config) {
+  dns_cache_config.set_name("dynamic_forward_proxy_cache_config");
+  if (GetParam() == Network::Address::IpVersion::v4) {
+    dns_cache_config.set_dns_lookup_family(envoy::config::cluster::v3::Cluster::V4_ONLY);
+  } else {
+    dns_cache_config.set_dns_lookup_family(envoy::config::cluster::v3::Cluster::V6_ONLY);
+  }
+}
+
+// Setup the bootstrap extension config to enable the internal listener feature.
+void HttpsInspectionIntegrationTest::setupBootstrapExtension() {
+  config_helper_.addBootstrapExtension(R"EOF(
+    name: envoy.bootstrap.internal_listener
+    typed_config:
+      "@type": type.googleapis.com/envoy.extensions.bootstrap.internal_listener.v3.InternalListener
+    )EOF");
+}
+
+void HttpsInspectionIntegrationTest::setupTransportSocket(envoy::config::listener::v3::FilterChain& filter_chain) {
+  if (!clear_http_test_) {
+    auto* transport_socket =  filter_chain.mutable_transport_socket();
```

Comment:
Is it possible to express all these changes in yaml? It is much easier to read than proto modifications in C++